locust_tasks/locustfile.py

```diff
@@ -176,15 +229,121 @@ class UserBehavior(TaskSet):
+#   - Optionally verifies that key content exists on the current page.
+#
+# In the event of failure it:
+#   - Reports key debugging information, such as step ID & the UAC, to aid with debugging.
+#   - For the error log it records the failure message and key content of the current page.
+#   - Aborts the current task.
+#
+def verify_response(id, task, resp, expected_status, expected_page, expected_content=''):
+    # print ('In verify_response(%s). Expected:%3d actual:%3d expected_page:%s' % (id, expected_status, resp.status_code, expected_page))
+    # print ('  URL:%s' % (resp.url))
+    # print ('  status:%d' % (resp.status_code))
+    # print ('  Expected page title:%s' % (expected_page.title))
+
+    # Sanity check for missing response 
+    if (resp.text in (None, '')):
+        failure_message = 'Expected to be on the ' + expected_page.name + ' page but got an empty response!'
+        report_failure(id, resp, task, failure_message, '')
+
+    # Page check
+    current_page = identify_page(id, task, resp, resp.text)
+    if (current_page != expected_page):
+        failure_message = 'On wrong page. Expected to be on ' + expected_page.name + ' page but am on ' + current_page.name + ' page.'
+        page_extract = extract_key_page_content(id, task, resp, current_page)
+        report_failure(id, resp, task, failure_message, page_extract)        
+    
+    # Status check
+    if (expected_status != resp.status_code):
+        failure_message = 'Status mismatch. Expected ' + str(expected_status) + ' but was ' + str(resp.status_code) + '.'
+        page_extract = extract_key_page_content(id, task, resp, current_page)
+        report_failure(id, resp, task, failure_message, page_extract)        
+    
+    # Content verification
+    if (expected_content not in (None, '')):
+        if (expected_content not in resp.text):
+            failure_message = current_page.name + ' page does not contain expected text (' + expected_content + ').'
+            page_extract = extract_key_page_content(id, task, resp, current_page)
+            report_failure(id, resp, task, failure_message, page_extract)
+    
+    resp.success()
+    
+#
+# Reports a test failure:
+#   - the error is reported to Locust
+#   - an error is logged with either whole or partial page content
+#
+def report_failure(id, resp, task, failure_message, page_content):
+    error_detail = ''
+    if (page_content not in (None, '')):
+        error_detail = ' Page content >>> ' + page_content + ' <<<'
+
+    resp.failure(f'ID={id} UAC={task.case["uac"]} Status={resp.status_code}: {failure_message}')
+    logger.error(f'ID={id} UAC={task.case["uac"]} Status={resp.status_code}: {failure_message}{error_detail}')
+    task.interrupt()
+
+# 
+# Identifies the current page based on its content.
+# It returns a Page enum value if the page can be identified, or fails the test if it cannot.
+#
+def identify_page(id, task, resp, page_content):
```

Comment:
remove page_content parameter. It is unused because you blat it on the next line!

Reply:
Deleted parameter